sycl/include/CL/sycl/experimental/spec_constant.hpp

```diff
@@ -32,7 +32,9 @@ template <typename T, typename ID = T> class spec_constant {
 private:
   // Implementation defined constructor.
 #ifdef __SYCL_DEVICE_ONLY__
+public:
   spec_constant() {}
+private:
```

Comment:
This change will violate the spec, won't it?

Reply:
I don't see how this violates the [spec](https://github.com/codeplaysoftware/standards-proposals/blob/master/spec-constant/index.md). Plus I do think that given the experimental and temporary status of the implemented spec we have certain degree of freedom to adjust it if we find it reasonable.